Fix #5273: UnusedAttribute Lint Warning

kavyanshkhandelwal opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
This PR addresses the `UnusedAttribute` lint warning in the codebase by removing unused attributes across multiple files. This change helps clean up the code and improve maintainability.

Fixes #5273: UnusedAttribute lint warning.
Deleted the Deprecated kotlin_imports_layout
Resolved the 'Use $(..) notation instead of legacy backticks `...`' warnings in the shell scripts.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

kavyanshkhandelwal committed:
(Issue #5273) Fix UnusedAttribute Lint Warning

kavyanshkhandelwal committed:
Merge branch 'develop' into kavyanshkhandelwal/#5273/fix-unused-attribute-warning

TanishMoral11 commented:
Hey @kavyanshkhandelwal , 
Welcome to Oppia! 
could you please make sure to fill out the CLA form? This is an important step for accepting your contributions.

Additionally, for a smoother review process, it would be great if you could include screenshots demonstrating the output before and after your changes. This will helps for verify the effectiveness of your fixes and ensure everything works as expected.

kavyanshkhandelwal commented:
> Hey @kavyanshkhandelwal , Welcome to Oppia! could you please make sure to fill out the CLA form? This is an important step for accepting your contributions.
> 
> Additionally, for a smoother review process, it would be great if you could include screenshots demonstrating the output before and after your changes. This will helps for verify the effectiveness of your fixes and ensure everything works as expected.


I have successfully filled out the CLA form. Here are the screenshots : 
![Before](https://github.com/user-attachments/assets/7bd73c8a-728b-4b0f-9deb-8a9e65cac27f)
![After](https://github.com/user-attachments/assets/c290c594-33c5-4e44-af98-45bbff56af54)

kavyanshkhandelwal committed:
Merge branch 'develop' into kavyanshkhandelwal/#5273/fix-unused-attribute-warning

kavyanshkhandelwal committed:
Fix linting issues using ktlint

adhiamboperes commented on gradlew:
```diff
@@ -11,21 +11,21 @@
         PRG="$link"
     else
-        PRG=`dirname "$PRG"`"/$link"
+        PRG=$(dirname "$PRG")"/$link"
     fi
 done
-SAVED="`pwd`"
-cd "`dirname \"$PRG\"`/" >/dev/null
-APP_HOME="`pwd -P`"
-cd "$SAVED" >/dev/null
+SAVED="$(pwd)"
+cd "$(dirname "$PRG")/" || exit 1
+APP_HOME="$(pwd -P)"
+cd "$SAVED" || exit 1
 
 APP_NAME="Gradle"
-APP_BASE_NAME=`basename "$0"`
+APP_BASE_NAME=$(basename "$0")
```
I don't understand how these changes are related to the unused attribute warning.

adhiamboperes commented on .idea/codeStyles/Project.xml:
```diff
@@ -5,8 +5,6 @@
       <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
     </JavaCodeStyleSettings>
     <JetCodeStyleSettings>
-      <option name="NAME_COUNT_TO_USE_STAR_IMPORT" value="2147483647" />
-      <option name="NAME_COUNT_TO_USE_STAR_IMPORT_FOR_MEMBERS" value="2147483647" />
```
Why were these deleted?

adhiamboperes commented on .idea/codeStyles/Project.xml:
```diff
@@ -161,4 +159,4 @@
     </codeStyleSettings>
   </code_scheme>
-</component>
+</component>
```
New line must be added at the end of every file.

adhiamboperes reviewed (changes requested):
Thanks @kavyanshkhandelwal, and apologies for the delayed review of your first PR.

In addition to the comments I have left inline:
1. Your PR description did not list the unused attributes that needed to be fixed.
2. I could not understand how the changes in the PR are solving the issue.(except the  kotlin_imports_layout` one) 
3. Please use before and after screenshots like [this](https://github.com/oppia/oppia-android/issues/5169#issuecomment-1855429949) on the PR description to show that the issue is actually resolved.
4. Please see [wiki here](https://github.com/oppia/oppia-android/wiki/Interpreting-CI-Results) for a guide on intepreting CI failures.

kavyanshkhandelwal committed:
Merge branch 'develop' into kavyanshkhandelwal/#5273/fix-unused-attribute-warning

kavyanshkhandelwal committed:
ActualFixingUnusedAttributeLintWarning

kavyanshkhandelwal commented:
Thanks @adhiamboperes for the PR review.
I apologize for the confusion earlier. I had mistakenly focused on resolving some issues related to shell script warnings, which were not part of the original issue. After your feedback, I’ve identified and made the necessary changes related to the unused attribute warnings.

Here’s a summary of the changes I made:

1) In "bottom_sheet_options_menu_fragment.xml", fixed "android:drawableTint to app:drawableTint" at 3 places.
2) In "circular_progress_indicator_adapters_test_activity.xml", fixed android:min by using tools:targetApi="26".
3) In "content_item.xml", fixed android:breakStrategy by using tools:targetApi="23".
4) In "feedback_item.xml", fixed android:breakStrategy by using tools:targetApi="23".
5) In "profile_list_control_buttons.xml", fixed android:min by using tools:targetApi="26".
6) In "profile_progress_recently_played_story_card.xml", fixed android:forceDarkAllowed by using tools:targetApi="29".
7) In "promoted_story_card.xml", fixed android:forceDarkAllowed by using tools:targetApi="29".
8) In "recently_played_story_card.xml", fixed android:forceDarkAllowed by using tools:targetApi="29".
9) In "resume_lesson_fragment.xml", changed android:drawableTint to appdrawableTint at 2 places.

Inside the Android -> Lint -> Correctness (showing error):
Changed the android:tint to app:tint in 5 XML files:
"add_profile_activity.xml"
"administrator_controls_activity.xml"
"drag_drop_interaction_items.xml"
"drawer_fragment.xml"
"help_activity.xml"

Unused Attributes Fixed :-
drawableTint
min
breakStrategy
forceDarkAllowed

Here are the screenshots of Code Inspect before and after the changes.
Screenshot Before Changes:
![Screenshot 2025-01-11 173819](https://github.com/user-attachments/assets/c095351b-e54f-4ce6-bc8d-ab8671832e95)

Screenshot After Changes:
![Screenshot 2025-01-11 225048](https://github.com/user-attachments/assets/0a117562-e258-4f3c-9616-6a3bc15b0fb6)

adhiamboperes commented:
Thanks for making these changes @kavyanshkhandelwal! I’ll review this tomorrow

adhiamboperes commented:
@kavyanshkhandelwal, did you forget to push a commit with your latest changes?

kavyanshkhandelwal committed:
Actual Fix Unused Attribute Lint Warning #5273

kavyanshkhandelwal committed:
Merge branch 'kavyanshkhandelwal/#5273/fix-unused-attribute-warning' of https://github.com/kavyanshkhandelwal/oppia-android-forked into kavyanshkhandelwal/#5273/fix-unused-attribute-warning

kavyanshkhandelwal commented:
> @kavyanshkhandelwal, did you forget to push a commit with your latest changes?

Hi @adhiamboperes, I realized I had forgotten to push my latest changes earlier. I’ve just pushed them now. Let me know if there’s anything else I need to address. Thanks for pointing it out!

kavyanshkhandelwal committed:
Merge branch 'develop' into kavyanshkhandelwal/#5273/fix-unused-attribute-warning

adhiamboperes reviewed (changes requested):
@kavyanshkhandelwal, at the top of the PR, it shows that there are `1,498` changed files. Is it possible that you re-formatted the whole project?

kavyanshkhandelwal commented:
> @kavyanshkhandelwal, at the top of the PR, it shows that there are `1,498` changed files. Is it possible that you re-formatted the whole project?

@adhiamboperes I may have accidentally done that, though I’m unsure. If it works, I can reset the branch and then reapply the necessary changes for the unused attribute only. Please let me know if that approach works or if there is any other way to do that.

adhiamboperes commented:
@kavyanshkhandelwal, my advice would be to:
1. checkout the latest develop branch
2.  create a clean branch from develop
3. Copy the relevant changes from your current branch
4. Close this PR and open a replacement.

We do not allow force pushes, but a rebase would have otherwise been the easiest fix.

kavyanshkhandelwal commented:
Thanks @adhiamboperes for your feedback. I’ve created a new PR from a clean branch to address the changes properly and avoid the unintentional reformatting. I’ll close this PR now.
